situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java

```diff
@@ -1,53 +1,93 @@
 package com.amazon.situp.parser.model;
 
-import com.amazon.situp.model.configuration.Configuration;
-import com.amazon.situp.parser.ConfigurationDeserializer;
-import com.amazon.situp.parser.validator.PipelineComponent;
-import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.amazon.situp.model.configuration.PluginSetting;
+import com.amazon.situp.parser.ParseException;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.String.format;
 
-@JsonDeserialize(using = ConfigurationDeserializer.class)
 public class PipelineConfiguration {
 
-    @PipelineComponent(type = PipelineComponent.Type.Source, message = "Invalid source configuration; Requires " +
-            "exactly one valid source")
-    private final Configuration source;
+    @NotNull(message = "source settings are missing in provided configuration")
+    private final PluginSetting sourcePluginSetting;
+    private final PluginSetting bufferPluginSetting;
+    private final List<PluginSetting> processorPluginSettings;
+    @NotEmpty(message = "sink settings are missing in provided configuration, atleast one sink is required")
+    private final List<PluginSetting> sinkPluginSettings;
+    private final int workers;
+    private final int readBatchDelay;
 
-    @PipelineComponent(type = PipelineComponent.Type.Buffer, message = "Invalid buffer configuration; Requires" +
-            " at most one valid buffer")
-    private final Configuration buffer;
+    @JsonCreator
+    public PipelineConfiguration(
+            @JsonProperty("source") final Map<String, Map<String, Object>> source,
+            @JsonProperty("buffer") final Map<String, Map<String, Object>> buffer,
+            @JsonProperty("processor") final Map<String, Map<String, Object>> processors,
+            @JsonProperty("sink") final Map<String, Map<String, Object>> sinks,
+            @JsonProperty("workers") final int workers,
+            @JsonProperty("delay") final int delay) {
+        this.sourcePluginSetting = getFirstPluginSettingFromConfiguration(source, "source");
+        this.bufferPluginSetting = getFirstPluginSettingFromConfiguration(buffer, "buffer");
+        this.processorPluginSettings = getAllPluginSettingsFromConfiguration(processors);
```

Comment:
Can we not instead of concrete methods? getSourceFromPluginSetting, getBuffer, getProcessors and getSinks?